File: easy/split_the_number.rb
File.open(ARGV[0]).each_line do |line|
  s = line.chomp.split
  next if s.empty?
  c = r = v = 0
  o = 1
  s[1].each_char do |i|
    if i =~ /[a-z]/
      v = v * 10 + s[0][c].to_i
      c += 1
    elsif i == '+'
      r += v * o
      o = 1
      v = 0
    elsif i == '-'
      r += v * o
      o = -1
      v = 0
    end
  end
  puts r + v * o
end

File: easy/split_the_number.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    next if ( $line =~ m/^\s$/ );
    chomp($line);
    my ( $l, $r ) = split( / /, $line );
    my ( $c, $s, $v, $o ) = ( 0, 0, 0, 1 );
    map {
        if ( $_ =~ /[a-z]/ )
        {
            $v = $v * 10 + substr( $l, $c, 1 );
            $c++;
        }
        elsif ( $_ eq '+' ) {
            $s += $v * $o;
            ( $v, $o ) = ( 0, 1 );
        }
        elsif ( $_ eq '-' ) {
            $s += $v * $o;
            ( $v, $o ) = ( 0, -1 );
        }
    } split( //, $r );
    printf( "%d\n", $s + $v * $o );
}
close(INFILE);
